src/irmin/atomic_write.ml

```diff
@@ -17,65 +17,45 @@
 open Import
 include Atomic_write_intf
 
-module Check_closed (Make_atomic_write : Maker) (K : Type.S) (V : Type.S) =
-struct
-  module S = Make_atomic_write (K) (V)
-
-  type t = { closed : bool ref; t : S.t }
-  type key = S.key
-  type value = S.value
+module Check_closed_store (AW : S) = struct
+  type t = { closed : bool ref; t : AW.t }
+  type key = AW.key
+  type value = AW.value
+  type watch = AW.watch
 
-  let check_not_closed t = if !(t.closed) then raise Store_properties.Closed
+  let make_closeable t = { closed = ref false; t }
 
-  let mem t k =
-    check_not_closed t;
-    S.mem t.t k
+  let get_open_exn t =
```

Comment:
`get_open_exn` is a little confusing, maybe `store_if_open_exn` or `get_if_open_exn` would be more clear?

Reply:
Yeah, I just copied the existing function but I think `get_if_open_exn` is a better name. I'll make this change. :+1: